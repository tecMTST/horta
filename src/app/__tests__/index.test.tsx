import * as React from 'react';
import { render } from '@testing-library/react';
import axios from 'axios';

import { App } from '../index';

// Mock jest and set the type
jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('<App />', () => {
  mockedAxios.get.mockResolvedValueOnce({});
  const renderComponent = () => render(<App />);

  it('should render and match the snapshot', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });
});
