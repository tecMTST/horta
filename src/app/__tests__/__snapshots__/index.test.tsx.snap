// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<BrowserRouter>
  <Helmet
    defaultTitle="React Boilerplate"
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={
      Object {
        "lang": undefined,
      }
    }
    prioritizeSeoTags={false}
    titleTemplate="%s - React Boilerplate"
  >
    <meta
      content="A React Boilerplate application"
      name="description"
    />
  </Helmet>
  <Routes>
    <Route
      element={<LoginPage />}
      path="/login"
    />
    <Route
      element={<PrivateRoute />}
      path="/"
    >
      <Route
        element={<HomePage />}
        path="/"
      />
    </Route>
  </Routes>
  <Memo(l) />
</BrowserRouter>
`;
